fluss-common/src/main/java/org/apache/fluss/metadata/Schema.java

```diff
@@ -614,7 +633,7 @@ private static List<Column> normalizeColumns(
                 newColumns.add(
                         new Column(
                                 column.getName(),
-                                column.getDataType().copy(false),
+                                column.getDataType().copy(true),
```

Comment:
Partial update is required, so the data type here needs to be temporarily set to nullable.